app/helpers/discount-value.js

```diff
@@ -1,19 +1,21 @@
 import Helper from '@ember/component/helper';
 
-export function discountValue(params) {
-  let [attendee] = params;
-  const { ticket, order } = attendee;
-  if (order.get('discountCode') && ticket.get('discountCodes')) {
-    discountCode = order.get('discountCode');
-    ticket.get('discountCodes').forEach(discount => {
-      if(discount.get('code') === discountCode.get('code')) {
-        if (discount.get('type') === 'amount') {
-          return discount.get('value');
-        } else {
-          return (ticket.get('price') * discount.get('value'))/100;
+export async function discountValue(params) {
```

Comment:
Don't make helpers async. Also, so much business logic in a helper is a bad idea. Do all this in the controller before you pass the data down to the template.

Reply:
@niranjan94 actually I need to calculate discount for each entry (row). The component `cell-price.js` doesn't return even basic values to the template. 
I don't think we can use controller here as we need to calculate the values corresponding to each row dynamically and controller will be executed only once.

Reply:
@shreyanshdwivedi the rows are coming in from the route -> controller -> component right ? Why not make sure the rows already come in with the calculated discount amount ? It is possible. 

Never misuse helpers like this. They are not meant for such things.